### ISSUE TITLE
Add limits on development platform when editing submissions as an admin

### ISSUE THREAD

**Issue by dboyer** [Iridescent-CM/technovation-app#3372]
Split off from #3234 as this is not as time sensitive. All tasks in this issue pertain to the admin page that supports editing submissions. Go to /admin/team_submissions#/ and then select a submission to view the page where admins can edit submitted fields.

- [ ] I selected Mobile App as my submission type and app inventor as my development platform. I saved and it forced me to enter something in the app inventor app name field, which is correct behavior. However, I then entered a URL in the Thunkable project URL field and was able to save. The view after saving did not display the Thunkable URL, which I'm guessing is because "app inventor" was selected as my development platform. It would be helpful if we could have the page return a warning:

- If app inventor selected as the development platform and an admin tries to add a Thunkable project URL, show an error on the Thunkable project URL field saying "Cannot add a Thunkable project URL when App Inventor selected as the development platform." Do not allow user to save the form with those edits
- If Thunkable selected as the development platform and an admin tries to add an App Inventor app name, show an error on the App Inventor name field saying "Cannot add an App Inventor app name when Thunkable selected as the development platform." Do not allow user to save the form with those edits.

<img width="426" alt="Screen Shot 2022-04-25 at 8 18 59 PM" src="https://user-images.githubusercontent.com/516764/165194816-c264aa6a-cf39-447c-9110-da24746fa6bc.png"> 

<img width="438" alt="Screen Shot 2022-04-25 at 8 19 49 PM" src="https://user-images.githubusercontent.com/516764/165194892-29ac30bb-96c1-4fa8-bca6-6a1f3c7797af.png">
 

- [ ] When editing the pitch video and demo video fields, it is possible to put text in the field rather than a URL. Require a link to a URL. 
<img width="440" alt="Screen Shot 2022-05-03 at 10 18 50 AM" src="https://user-images.githubusercontent.com/516764/166470981-f48e0b46-5d00-4361-ad22-0e17b9df590d.png">

**Comment by devkonrad** [comment]
Created the PR https://github.com/Iridescent-CM/technovation-app/pull/3412 to deal with this demand.

**Comment by viviancan** [comment]
` When editing the pitch video and demo video fields, it is possible to put text in the field rather than a URL. Require a link to a URL.`

@dboyer Following up one of the checkboxes - is the word "link" supposed to show instead of the video URL?

**Comment by viviancan** [comment]
A few followup notes - 

- What is considered a "valid" video URL? 

- When I changed a submission to thunkable and tried to save it with a valid thunkable project URL I could not successfully save the edited submission. I even tried saving it with a thunkable email too and same issue. ![Screen Shot 2022-05-20 at 3.21.06 PM.png](https://images.zenhubusercontent.com/606b0f4fd13d53dff3d14fa5/e631f757-131a-4ce2-9687-c8a2a3c06871)

**Comment by devkonrad** [comment]
> A few followup notes -
> 
> * What is considered a "valid" video URL?
> * When I changed a submission to thunkable and tried to save it with a valid thunkable project URL I could not successfully save the edited submission. I even tried saving it with a thunkable email too and same issue. ![Screen Shot 2022-05-20 at 3.21.06 PM.png](https://camo.githubusercontent.com/3210dae00a11eb2eb0e0c65d0d56e6526602a1cffcbb70e6a936ed75f8bfdadf/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3630366230663466643133643533646666336431346661352f65363331663735372d313331612d346365322d393638372d633861326133633036383731)

Hi @viviancan as I could note about this validation. The app was considering valid URL the pattern: "https://x.thunkable.com/projects/something"

**Comment by dboyer** [comment]
> ` When editing the pitch video and demo video fields, it is possible to put text in the field rather than a URL. Require a link to a URL.`
> 
> @dboyer Following up one of the checkboxes - is the word "link" supposed to show instead of the video URL for pitch and demo?

Currently, it's possible to just type text into the pitch or demo fields. Only URLs should be allowed, and it is fine for the URL to show. When @devkonrad demoed from his local machine, it looked like that was working as expected now.

**Comment by devkonrad** [comment]
> > ` When editing the pitch video and demo video fields, it is possible to put text in the field rather than a URL. Require a link to a URL.`
> > 
> > @dboyer Following up one of the checkboxes - is the word "link" supposed to show instead of the video URL for pitch and demo?
> 
> Currently, it's possible to just type text into the pitch or demo fields. Only URLs should be allowed, and it is fine for the URL to show. When @devkonrad demoed from his local machine, it looked like that was working as expected now.

Thats right, testing locally the issue. Just URL are allowed on the pitch and demo link fields.

**Comment by devkonrad** [comment]
> A few followup notes -
> 
> * What is considered a "valid" video URL?
> * When I changed a submission to thunkable and tried to save it with a valid thunkable project URL I could not successfully save the edited submission. I even tried saving it with a thunkable email too and same issue. ![Screen Shot 2022-05-20 at 3.21.06 PM.png](https://camo.githubusercontent.com/3210dae00a11eb2eb0e0c65d0d56e6526602a1cffcbb70e6a936ed75f8bfdadf/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3630366230663466643133643533646666336431346661352f65363331663735372d313331612d346365322d393638372d633861326133633036383731)

Hey, @viviancan ! I just fixed the validation for the thunkable project link url. Here is the PR: https://github.com/Iridescent-CM/technovation-app/pull/3412

**Comment by devkonrad** [comment]
> Following up here, I have already added the comment to the code review! The changes for the URL validator that includes the "project" version of the thunkable URL is the correct version. This will need to be reverted before merging to QA. Apologies for any confusion around the issue!

No problem, @viviancan ! It happens :) I just reverted the changes and pushed the PR again.
